src/orion/core/utils/random_state.py

```diff
@@ -77,3 +78,22 @@ def seed(cls, base_seed: int | None) -> RandomState:
             # Add the base_seed property in this case.
             random_state = replace(random_state, base_seed=base_seed)
         return random_state
+
+
+@contextmanager
+def control_randomness(random_state: RandomState | None):
+    """Seeds the randomness inside the indented block of code using `self.random_state`."""
+    if random_state is None:
+        yield
+        return
+
+    # Save the initial random state.
+    initial_rng_state = RandomState.current()
+    # Set the random state.
+    random_state.set()
```

Comment:
Could move the if statement from above down here:
```suggestion
    if random_state:
        random_state.set()
```

Reply:
Would also need to add it after the yield. I think it's simpler to have it as a first check.